DataDog/datadog-operator
# [profiles] Add label overrides for profiles

### What does this PR do?

Add label overrides for profiles

### Motivation

https://datadoghq.atlassian.net/browse/CECO-1311

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: n/a
* Cluster Agent: n/a

### Describe your test plan

- Deploy the operator with profiles enabled
- Create a DDA with label overrides for the node agent similar to the example below
```yaml
  override:
    nodeAgent:
      labels:
        "foo": "bar"
        "one": "two"
```
- Create a profile with label overrides for the node agent, adding at least 1 new label and replacing at least 1 label
```yaml
    override:
      nodeAgent:
        labels:
          "foo": "foo"
          "bar": "bar"
```
- Check for the expected labels in the resulting agent pods and daemonsets. In this example, the default daemonset should have `foo:bar` and `one:two`. The profile daemonset should have `foo:foo`, `one:two`, and `bar:bar`
```
kubectl get pods --show-labels
kubectl get ds --show-labels
```
- Update the profile with new and updated labels
```yaml
    override:
      nodeAgent:
        labels:
          "foo": "foo2"
          "bar": "bar"
          "three": "four"
```
- Check the labels were updated. In this example, the default daemonset should have `foo:bar` and `one:two`. The profile daemonset should have `foo:foo2`, `one:two`, `bar:bar`, and `three:four`


### Checklist
